pkg/ccl/streamingccl/streamclient/cockroach_sinkless_replication_client.go

```diff
@@ -123,9 +124,15 @@ func (m *sinklessReplicationClient) ConsumePartition(
 				return
 			}
 		}
-		if err := rows.Err(); err != nil {
+		if err := rows.Err(); errors.Is(err, driver.ErrBadConn) {
```

Comment:
mmm this seems a little wrong, it should probably be:

```
if err := rows.Err(); err != nil {
  // Maybe add a comment about why we are special casing a client disconnect.
  if errors.Is(err, driver.ErrBadConn) {
       select {
          case eventCh <- streamingccl.MakeGenerationEvent():
          case <-ctx.Done():
				       errCh <- ctx.Err()
			 }
  } else {
      errCh <- err
  }
}
```
Let's chat offline about how `select` blocks and channels work, just to make sure we're on the same page 🙂